pkg/daemon/daemon.go

```diff
@@ -2176,26 +2178,59 @@ func (dn *Daemon) completeUpdate(desiredConfigName string) error {
 	return nil
 }
 
+func (dn *Daemon) revertToNonLayeredState(desiredConfig *mcfgv1.MachineConfig, currentImage string) error {
+	// Get the default OS image from the machine-config-osimageurl ConfigMap or the machine-config-images ConfigMap.
+	defaultOSImage, _ := dn.loadFromConfigMap()
+	return dn.updateImage(nil, desiredConfig, currentImage, defaultOSImage, false)
+}
+
+// loadFromConfigMap retrieves the default OS image URL from a ConfigMap.
+func (dn *Daemon) loadFromConfigMap() (string, error) {
+	// Setup Kubernetes client
+	config, err := rest.InClusterConfig()
```

Comment:
thought: We can simplify and / or consolidate access to this ConfigMap.

If we modify the DaemonSet manifest for the MCD, we can attach this ConfigMap to the pod as a volume. In that case, reading the ConfigMap would be like reading a local file. Also, this isn't the only place we need to read from or decode the JSON contained within this ConfigMap so it could make sense to hoist this function into something like `ctrlcommon` or the like. Both of those suggestions are out-of-scope for this PR though.